base_layer/core/src/base_node/state_machine_service/states/header_sync.rs

```diff
@@ -112,8 +113,13 @@ impl HeaderSyncState {
         });
 
         let local_nci = shared.local_node_interface.clone();
-        synchronizer.on_rewind(move |blocks| {
-            local_nci.publish_block_event(BlockEvent::BlockSyncRewind(blocks));
+        synchronizer.on_rewind(move |removed| {
+            if let Some(fork_height) = removed.last().map(|b| b.height()) {
```

Comment:
Are we sure `removed` is ordered by height ASC?
I think it's ordered by height DESC. 
But I don't think anywhere we guarantee or mention this. This might be an oversight.

Reply:
It's ordered ASC (tip to fork height), this is documented in the `rewind_to_height` function in blockchain database. Agree it's not super clear.

Reply:
We don't currently test metrics in cucumber. Updated the description.